docs/RationaleSeq.md

```diff
@@ -0,0 +1,100 @@
+# Seq(uential) Dialect Rationale
+
+This document describes various design points of the `seq` dialect, why it is
+the way it is, and current status. This follows in the spirit of other [MLIR
+Rationale docs](https://mlir.llvm.org/docs/Rationale/).
+
+## Introduction
+
+Digital logic is generally split into two orthogonal categories:
+combinational and sequential. CIRCT contains a `comb` dialect to model the
+basic combinational operations and the (future) `seq` dialect which is
+discussed here. The intention of the `seq` dialect is to provide a set of
+stateful constructs which can be used to model sequential logic, independent
+of the output method (e.g. SystemVerilog).
+
+We have yet to flesh this dialect out to represent all common stateful
+components. We start with a register and will build from there.
+
+## Definitions
+
+For the sake of precision, we use the following definitions:
+
+- Physical devices:
+  - **Unclocked Latch:** A memory element which is only sensitive to the
+  levels of its inputs. Has no clock. Example: SR Latch.
+  - Clocked (gated) **latch:** A latch wherein the inputs are gated by a
+  clock. Transparent the entire time the clock is high. Generally referred to
+  as a "latch". Examples: "gated SR latch", "D latch".
+  - Edge-triggered **flip-flop:** An edge-sensitive memory element. Captures
+  the input value on one or both clock edges. Variants: posedge FF, negedge
+  FF, "edge-sensitive" FF (captures the input value on both edges),
+  resettable FF.
+- Abstract models:
+  - **Register:** A synchronous, resettable memory element. Can be
```

Comment:
Maybe say no intrinsic sub-addressing?  Setting/reading is as a whole.

Reply:
I thought that was implied, but I can add that note.